ls1intum/Artemis
# Improved Course-card component including test

### Checklist
- [x] Client: I updated the integration test (Jest) related to the features (with a high test coverage)

### Motivation and Context

Currently, the **Next Exercise**, which is displayed in a **course card** (on the main course overview page) is simply the exercise that has the **soonest (smallest) Due Date**.
![image](https://user-images.githubusercontent.com/65814168/116921521-62380180-ac54-11eb-8655-386b38b27136.png)
![image](https://user-images.githubusercontent.com/65814168/116921832-d96d9580-ac54-11eb-8ad5-23d38a6c3f87.png)
This PR makes sure, that the next Excercise displayed does **not** yet have 100% (not successful).
![image](https://user-images.githubusercontent.com/65814168/116922111-2d787a00-ac55-11eb-911a-bdc77d065a50.png)


Fixes #2583 -> Core Motivation: Why display exercises that the student already completed with 100%? -> Redundant Information -> Better to promote open/not-yet-successful excercises.
### Description

Simply filters out the exercises that are already successful, therefore shows the exercise with the soonest/shortest/earliest due date, that still needs to be worked on.

### Steps for Testing

1. Log in to Artemis
2. Create or find (at least) two exercises 
    - where both deadlines are in the future 
    - both need to have different due dates
    -  both are not yet successful (< 100%)
    - _(preferably text exercises, because they seem to be easier/faster)_
3. Go to the main/start page (course overview)
    -  make sure that the exercise with the earliest due date is shown
4. Complete the exercise with the earliest due date and correct it (give it 100%, make sure it is successful)
5. Go back to the main/start page (course overview)
    - now the exercise with the earliest due date that was shown prior should be replaced by another one that has a later due date but is not yet at 100%
    

p.s.  This is my very first PR, so feel free to point things out that I can improve upon.

Big Thank you to Simon E. and Nicolas R. for helping me out with getting to know Artemis and its developer workflow!
